source/common/cardEeprom.c

```diff
@@ -97,22 +97,35 @@ uint32 cardEepromGetSize() {
 	if(type == 1)
 		return 512;
 	if(type == 2) {
-		u32 buf1,buf2,buf3;
+		u32 buf1,buf2,buf3 = 0x54534554; // "TEST"
+		// Save the first word of the EEPROM
 		cardReadEeprom(0,(u8*)&buf1,4,type);
-		if ( !(buf1 != 0 || buf1 != 0xffffffff) ) {
-			buf3 = ~buf1;
-			cardWriteEeprom(0,(u8*)&buf3,4,type);
-		} else {
-			buf3 = buf1;
-		}
+
+		// Write "TEST" to it
+		cardWriteEeprom(0,(u8*)&buf3,4,type);
+
+		// Loop until the EEPROM mirrors and the first word shows up again
 		int size = 8192;
-		while (1) {	 
```

Comment:
Might want to put a size cap here, or inside the loop. Don't know what value would be sensible, but in theory one could make a hostile card implementation that causes an infinite loop.

Reply:
Ah, I didn't see your comment before merging. Yes, it would be nice to have a cap.